common/sgx/verifier.c

```diff
@@ -579,34 +578,33 @@ static oe_result_t _verify_evidence(
 
     if (!memcmp(format_id, &_local_uuid, sizeof(oe_uuid_t)))
     {
-        // evidence_buffer has oe_report_header_t header,
-        // followed by an SGX report for local attestation.
-        oe_report_header_t* report = (oe_report_header_t*)evidence_buffer;
+        // evidence_buffer has an SGX report for local attestation
+        // followed by an optional custom claims buffer.
+        // Note: sgx_report_t has no field that can be checked quickly
+        // to verify it being an SGX report.
 
-        if (evidence_buffer_size < sizeof(*report) ||
-            report->version != OE_REPORT_HEADER_VERSION ||
-            report->report_type != OE_REPORT_TYPE_SGX_LOCAL)
+        if (evidence_buffer_size < sizeof(sgx_report_t))
```

Comment:
There are many issues in the approach:
1. Older reports will not be able to pass the verification. Versioning control has to be enforced here. I.e.  
if (version == 1)
   ...
else if (version == 2)
   ...
else
   ...

Also, we cannot rely on  `version` field in `sgx_report_t` because OE does not own that structure.

Reply:
`oe_report_t` and `oe_attestation_header_t` are two different types of headers, though they both have a common `version` field with different values. `oe_attestation_header_t` is handled by the OE SDK attestation framework layer and is TEE-agnostic. `oe_report_t` is SGX-specific legacy header and handled by SGX plugins.

@yentsanglee I agree with you that it's better to have version control in header inspection.
With further consideration, I think we should keep the `oe_report_header_t` for evidence formats `OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION` and `OE_FORMAT_UUID_SGX_ECDSA`, and keep the `oe_attestation_header_t` at current version number (2).